gammapy/spectrum/fit.py

```diff
@@ -19,108 +18,44 @@ class SpectrumFit(object):
+    translating gammapy classes to sherpa classes and handling various aspects
+    of the fitting correctly. For usage examples see :ref:`spectral_fitting`
 
     Parameters
     ----------
-    obs_list : SpectrumObservationList
+    obs : SpectrumObservationList
         Observations to fit
-
-    Examples
-    --------
-
-    Example how to run a spectral analysis and have a quick look at the results.
-
-    ::
-
-        from gammapy.spectrum import SpectrumObservation, SpectrumFit
-        import matplotlib.pyplot as plt
-
-        filename = '$GAMMAPY_EXTRA/datasets/hess-crab4_pha/pha_obs23523.fits'
-        obs = SpectrumObservation.read(filename)
-
-        fit = SpectrumFit(obs)
-        fit.run()
-        fit.result.plot_fit()
-        plt.show()
-
-    TODO: put output image in gammapy-extra and show it here.
+    model : `~gammapy.spectrum.models.SpectralModel`, `~sherpa.models.ArithmeticModel`
+        Model to be fit
+    stat : str, `~sherpa.stats.Stat` 
+        Fit statistic to be used
     """
-
     FLUX_FACTOR = 1e-20
     """Numerical constant to bring Sherpa optimizers in valid range"""
     DEFAULT_STAT = 'wstat'
     """Default statistic to be used for the fit"""
-    DEFAULT_MODEL = models.PowerLaw(index=2 * u.Unit(''),
-                                    amplitude=1e-11 * u.Unit('cm-2 s-1 TeV-1'),
-                                    reference=1 * u.TeV)
-    """Default model to be used for the fit"""
-
-    def __init__(self, obs_list, stat=DEFAULT_STAT, model=DEFAULT_MODEL):
-        if isinstance(obs_list, SpectrumObservation):
-            obs_list = SpectrumObservationList([obs_list])
-        if not isinstance(obs_list, SpectrumObservationList):
-            raise ValueError('Wrong input format {}\nUse SpectrumObservation'
-                             'List'.format(type(obs_list)))
-
-        self.obs_list = obs_list
+
+    def __init__(self, obs, model, stat=DEFAULT_STAT):
+        if isinstance(obs, SpectrumObservation):
+            obs = [obs]
+
+        self.obs = SpectrumObservationList(obs)
         self.model = model
         self.statistic = stat
         self._fit_range = None
         from gammapy.spectrum import SpectrumResult
         # TODO : Introduce SpectrumResultList or Dict
         self._result = list()
-        for obs in obs_list:
-            self._result.append(SpectrumResult(obs=obs))
-
-    @classmethod
-    def from_pha_list(cls, pha_list):
-        """Create `~gammapy.spectrum.SpectrumFit` from a list of PHA files
-
-        Parameters
-        ----------
-        pha_list : list
-            list of PHA files
-        """
-        obs_list = SpectrumObservationList()
-        for temp in pha_list:
-            f = str(make_path(temp))
-            val = SpectrumObservation.read_ogip(f)
-            val.meta.phafile = f
-            obs_list.append(val)
-        return cls(obs_list)
+        for _ in self.obs:
+            self._result.append(SpectrumResult(obs=_))
```

Comment:
So result is sometimes a list of SpectrumResult objects?
The `result` property docstring says it's a single `SpectrumResult` object.
That's confusing.

If you sometimes need a list, maybe just always make it a list and have one entry for the case of one obs?